### PR TITLE
models obfuscation fix

### PR DESCRIPTION
**Pull request by VysotskiVadim**
### Description
R8 optimisations break serialization of mapbox-java models with unrecognised fields. See the case in 77bb24a7b6ec9bbd43e65751c8d319f232bf2d76 .

<del>I add rule to navigation instead of mapbox-java, because we distribute mapbox-java as a `jar` and it can't carry android specific obfuscation rules for consumers. </del>

### CODE REVIEW

**Comment by dzinad** [comment]
It's not going to help in this particular case (when auto-value-gson-runtime comes from mapbox-java), but maybe we should also update this file? https://github.com/mapbox/auto-value-gson/blob/main/auto-value-gson-runtime/src/main/resources/META-INF/proguard/autovaluegson.pro
For projects that use a direct dependency from auto-value-gson-runtime.

File: libnavigation-core/proguard-rules.pro [hunk 1]
```diff
@@ -6,3 +6,8 @@
 
 # --- com.mapbox.api.directions.v5.MapboxDirections ---
 -dontwarn com.sun.xml.internal.ws.spi.db.BindingContextFactory
+
+-keepclassmembers class com.mapbox.auto.value.gson.SerializableJsonElement {
```
[the remark this answers is not in the document]
Comment (VysotskiVadim) [reply libnavigation-core/proguard-rules.pro:10]: they are needed to customise java serialization, like here https://stackoverflow.com/questions/7290777/java-custom-serialization

**Comment by VysotskiVadim** [comment]
https://github.com/mapbox/auto-value-gson/pull/3

**Comment by VysotskiVadim** [comment]
> Will this .pro file be considered by R8 in the Nav SDK?

Good question! I tested using a direct dependency on specific version from local maven. For some reason I'm sure that it works with transitive dependencies, otherwise the feature of distributing `pro` files by libs developers just wouldn't make any sense. 
Do you think it makes sense to retest with a transitive dependency?

**Pull request merged**